### DISCUSSION TITLE
Request: uutils repo for awk, sed, and grep

### DISCUSSION THREAD

**Discussion by tgross35** [uutils/coreutils#10755] [Ideas]
Hello uutils team,

I know there has been some interest in implementing these tools in previous issues. Would you consider setting up a repository with basic structure? The three tools could likely be combined into a single repository since there is a lot of common functionality among them.

They will need a BRE and ERE regex engine. One of the maintainers at rust-lang/regex [stated that they would be open to adding a BRE/ERE parser to that crate](https://github.com/rust-lang/regex/issues/858#issuecomment-1125329824), so it would be a good idea to build that regex functionality independent of the rest of the implementation so that it could be added back to the create. Alternatively, that part of the functionality could begin as a PR against rust-lang/regex.

Extra: I would like to make the mild suggestion to license _only this new part_ of the project under Apache 2.0, which is somewhat common among Rust projects. I know this topic has been contentious so if there's more than, say, 8 messages regarding it then I'd say just stick with MIT. However, couple of merits:
- Retain the ability to be used in closed-source distributions like unix, macOS, etc (like MIT)
- Add that anybody who modifies the software must state what they modified, but does not have to provide source code. This means that they can still closed source distribute (like MIT) but anybody making changes at least has some incentive to commit improvements back to this repository (one of the big gripes by everyone who wanted GPL).
- Continues to allow this project's usage of libraries that are licensed under Apache, and actually covers that case better.

^ That's about all I have or ever want to have to say about that

Anyway, if you get the time to just outline what success for a grep/awk/sed tool looks like, then some of us could start working on it. Thanks!

**Comment by cakebaker** [comment]
For `sed` a repo has been added in the meantime: https://github.com/uutils/sed

**Reply by sylvestre** [reply]
and happy to welcome awk & grep

**Reply by oech3** [reply]
OK to close this thread now?

**Comment by oech3** [comment]
https://github.com/uutils/awk

**Comment by cakebaker** [comment]
And finally: https://github.com/uutils/grep
